compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
-
-      virtual ValueConstraint *allocate(int32_t key)
-         {
-         return _vp->createValueConstraint(key, NULL, NULL);
-         }
-      virtual void free(ValueConstraint *vc)
-         {
-         _vp->freeValueConstraint(vc);
-         }
-      virtual ValueConstraint *copy(ValueConstraint *vc)
-         {
-         Relationship *rel = _vp->copyRelationships(vc->relationships.getFirst());
-         StoreRelationship *storeRel = _vp->copyStoreRelationships(vc->storeRelationships.getFirst());
-         ValueConstraint *newvc = _vp->createValueConstraint(vc->getValueNumber(), rel, storeRel);
-         return newvc;
-         }
-
-      virtual TR::Compilation *comp() {return _vp->comp();}
-
+      void setVP (TR::ValuePropagation * vp);
+      virtual ValueConstraint * allocate (int32_t key);
+      virtual void free (ValueConstraint * vc);
+      virtual ValueConstraint * copy (ValueConstraint * vc);
+      virtual TR::Compilation * comp ();
       private:
       TR::ValuePropagation *_vp;
       };
 #else
    class ValueConstraintHandler
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
This can be changed to void setVP(OMR::ValuePropagation *vp) {_vp = vp;}. As long as we pass in the right ValuePropagation instance to this method, calling its virtual functions should dispatch to the right methods for a given project. Am I missing something?